benchmark.js

```diff
@@ -390,11 +401,33 @@ var Benchmark = (function() {
         if (!storage.running) {
           return;
         }
-        var implKey = "com/sun/midp/lcdui/DisplayDevice.gainedForeground0.(II)V";
-        var originalFn = Native[implKey];
-        Native[implKey] = function() {
-          startup.stopTimer();
-          originalFn.apply(null, arguments);
+
+        var vmImplKey = "com/sun/midp/main/MIDletSuiteUtils.vmEndStartUp.(I)V";
+        var vmOriginalFn = Native[vmImplKey];
+        var vmCalled = false;
+        Native[vmImplKey] = function() {
+          if (!vmCalled) {
+            vmCalled = true;
+            startup.stopTimer("vmStartupTime");
+            startup.startTimer("backgroundStartupTime");
```

Comment:
Nit: since *startTimer* and *stopTimer* both call *performance.now*, a tiny bit of *totalStartupTime* will be unaccounted for in its constituent parts. It's probably insignificant, but I would still endeavour to use the same value when stopping the previous partial timer as I do when starting the next one, f.e. by calling *performance.now* here and passing its value to both functions.